package/kernel/rtl8373/Makefile

```diff
@@ -0,0 +1,24 @@
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
+
+PKG_NAME:=switch-rtl8373
+PKG_RELEASE:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define KernelPackage/switch-rtl8373
+  SECTION:=kernel
+  CATEGORY:=Kernel modules
+  SUBMENU:=Network Devices
+  TITLE:=Realtek RTL8373 switch support
+  DEPENDS:=+kmod-i2c-gpio
+  FILES:= $(PKG_BUILD_DIR)/rtl8373.ko
```

Comment:
I didn't find any information about this switch. ref: https://lore.kernel.org/all/?q=rtl8373

Reply:
There isn't any driver not publicly available datasheet for this family of RealTek switch ICs.
RTL8272 and RTL8273 are supposedly fairly similar, they contain an 8051 CPU, but can also be used as standalone dump switches. They can *not* run Linux themselves. However, one can implement a firmware on the 8051 CPU which allows Linux running on a connected SoC to manage the switch via MDIO.

There have been some efforts to reverse-engineer the firmware running on the DW8051 core on a standalone switch:
https://github.com/up-n-atom/SWTG118AS/issues/4